File: src/main/java/com/hedvig/backoffice/graphql/types/Transaction.java
package com.hedvig.backoffice.graphql.types;

import java.time.Instant;
import java.util.UUID;
import javax.money.MonetaryAmount;
import lombok.Value;

@Value
public class Transaction {
  UUID id;
  MonetaryAmount amount;
  Instant timestamp;
  String type;
  String status;

  public static Transaction fromDTO(com.hedvig.backoffice.services.payments.dto.Transaction dto) {
    return new Transaction(dto.getId(), dto.getAmount(), dto.getTimestamp(), dto.getType(),
        dto.getStatus());
  }
}

File: src/main/java/com/hedvig/backoffice/graphql/types/Transaction.kt
package com.hedvig.backoffice.graphql.types

import com.hedvig.graphql.commons.type.MonetaryAmountV2
import java.time.Instant
import java.util.*
import com.hedvig.backoffice.services.payments.dto.Transaction as TransactionDto

data class Transaction(
  var id: UUID,
  var amount: MonetaryAmountV2,
  var timestamp: Instant,
  var type: String,
  var status: String
) {
  companion object {
    fun fromDTO(dto: TransactionDto) = Transaction(
      id = dto.id,
      amount = MonetaryAmountV2.of(dto.amount),
      timestamp = dto.timestamp,
      type = dto.type,
      status = dto.status
    )
  }
}
